### PR TITLE
 CB-11578 fix some calls which used api level 16

### PR DESCRIPTION
**Pull request by DrMoriarty**
### Platforms affected
Android

### What does this PR do?
I fixed some crashes on devices with android 4.0 (API level 15 and lower)

### What testing has been done on this change?
run project with this plugin using

### Checklist
- [ ] [ICLA](http://www.apache.org/licenses/icla.txt) has been signed and submitted to secretary@apache.org.
- [ ] [Reported an issue](http://cordova.apache.org/contribute/issues.html) in the JIRA database
- [ ] Commit message follows the format: "CB-3232: (android) Fix bug with resolving file paths", where CB-xxxx is the JIRA ID & "android" is the platform affected.
- [ ] Added automated test coverage as appropriate for this change.

### CODE REVIEW

**Comment by cordova-qa** [comment]
Cordova CI Build has completed successfully.

**Commit**     - [Link](https://github.com/apache/cordova-plugin-inappbrowser/pull/170/commits/450f9415d2583bb241539dfba6648dfaa6e0a052)
**Dashboard** - [Link](http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-inappbrowser-pr/41/)

| Builder Name  | Console Output | Test Report | Device Logs  |
|     :---:     |     :---:      |   :---:     |     :---:    |
| [Windows 8.1 Store]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-inappbrowser-pr/41//PLATFORM=windows-8.1-store/) | [Link]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-inappbrowser-pr/41//PLATFORM=windows-8.1-store/console) | [Link]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-inappbrowser-pr/41//PLATFORM=windows-8.1-store/testReport/) | [Link]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-inappbrowser-pr/41//PLATFORM=windows-8.1-store/artifact/) |
| [Windows 10  Store]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-inappbrowser-pr/41//PLATFORM=windows-10-store/) | [Link]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-inappbrowser-pr/41//PLATFORM=windows-10-store/console) | [Link]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-inappbrowser-pr/41//PLATFORM=windows-10-store/testReport/) | [Link]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-inappbrowser-pr/41//PLATFORM=windows-10-store/artifact/) |
| [Windows 8.1 Phone]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-inappbrowser-pr/41//PLATFORM=windows-8.1-phone/) | [Link]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-inappbrowser-pr/41//PLATFORM=windows-8.1-phone/console) | [Link]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-inappbrowser-pr/41//PLATFORM=windows-8.1-phone/testReport/) | [Link]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-inappbrowser-pr/41//PLATFORM=windows-8.1-phone/artifact/) |
| [iOS]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-inappbrowser-pr/41//PLATFORM=ios/) | [Link]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-inappbrowser-pr/41//PLATFORM=ios/console) | [Link]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-inappbrowser-pr/41//PLATFORM=ios/testReport/) | [Link]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-inappbrowser-pr/41//PLATFORM=ios/artifact/) |
| [Android]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-inappbrowser-pr/41//PLATFORM=android/) | [Link]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-inappbrowser-pr/41//PLATFORM=android/console) | [Link]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-inappbrowser-pr/41//PLATFORM=android/testReport/) | [Link]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-inappbrowser-pr/41//PLATFORM=android/artifact/) |

File: src/android/InAppBrowser.java [hunk 1]
```diff
@@ -579,11 +579,13 @@ public void run() {
                 Resources activityRes = cordova.getActivity().getResources();
                 int backResId = activityRes.getIdentifier("ic_action_previous_item", "drawable", cordova.getActivity().getPackageName());
                 Drawable backIcon = activityRes.getDrawable(backResId);
-                back.setBackground(null);
+                if (Build.VERSION.SDK_INT >= 16)
+                    back.setBackground(null);
                 back.setImageDrawable(backIcon);
```
Comment (jcesarmobile) [inline src/android/InAppBrowser.java]: Add else forward.setBackgroundDrawable(null);

File: src/android/InAppBrowser.java [hunk 2]
```diff
@@ -600,11 +602,13 @@ public void onClick(View v) {
                 forward.setId(Integer.valueOf(3));
                 int fwdResId = activityRes.getIdentifier("ic_action_next_item", "drawable", cordova.getActivity().getPackageName());
                 Drawable fwdIcon = activityRes.getDrawable(fwdResId);
-                forward.setBackground(null);
+                if (Build.VERSION.SDK_INT >= 16)
+                    forward.setBackground(null);
```
Comment (jcesarmobile) [inline src/android/InAppBrowser.java]: add else with back.setBackgroundDrawable(null);

File: src/android/InAppBrowser.java [hunk 3]
```diff
@@ -579,11 +579,13 @@ public void run() {
                 Resources activityRes = cordova.getActivity().getResources();
                 int backResId = activityRes.getIdentifier("ic_action_previous_item", "drawable", cordova.getActivity().getPackageName());
                 Drawable backIcon = activityRes.getDrawable(backResId);
-                back.setBackground(null);
+                if (Build.VERSION.SDK_INT >= 16)
+                    back.setBackground(null);
```
Comment (jcesarmobile) [inline src/android/InAppBrowser.java]: Add else with back.setBackgroundDrawable(null);

File: src/android/InAppBrowser.java [hunk 4]
```diff
@@ -644,11 +648,13 @@ public boolean onKey(View v, int keyCode, KeyEvent event) {
                 close.setId(Integer.valueOf(5));
                 int closeResId = activityRes.getIdentifier("ic_action_remove", "drawable", cordova.getActivity().getPackageName());
                 Drawable closeIcon = activityRes.getDrawable(closeResId);
-                close.setBackground(null);
+                if (Build.VERSION.SDK_INT >= 16)
+                    close.setBackground(null);
```
Comment (jcesarmobile) [inline src/android/InAppBrowser.java]: Add else with back.setBackgroundDrawable(null);

**Comment by jcesarmobile** [comment]
Thanks for the PR, looks good, but I've added a few comments. 
Can you add the modifications and change the title to start with CB-11578 so it's linked with the JIRA issue?

Thanks

File: src/android/InAppBrowser.java [hunk 5]
```diff
@@ -579,11 +579,13 @@ public void run() {
...
                 Drawable backIcon = activityRes.getDrawable(backResId);
-                back.setBackground(null);
+                if (Build.VERSION.SDK_INT >= 16)
+                    back.setBackground(null);
                 back.setImageDrawable(backIcon);
                 back.setScaleType(ImageView.ScaleType.FIT_CENTER);
                 back.setPadding(0, this.dpToPixels(10), 0, this.dpToPixels(10));
-                back.getAdjustViewBounds();
+                if (Build.VERSION.SDK_INT >= 16)
+                    back.getAdjustViewBounds();
```
Comment (scyclops) [inline src/android/InAppBrowser.java]: Out of curiosity, what purpose do the calls to getAdjustViewBounds serve?  When I read the documentation, it doesn't appear that calling getAdjustViewBounds has any side effects.

https://developer.android.com/reference/android/widget/ImageView.html#getAdjustViewBounds()

Comment (jcesarmobile) [inline src/android/InAppBrowser.java]: Yeah, I don't think it makes anything as it's a get, it returns true or false and the value is not being used anywhere or stored on a variable

**Comment by DrMoriarty** [comment]
@jcesarmobile   done

**Comment by cordova-qa** [comment]
Cordova CI Build has completed successfully.

**Commit**     - [Link](https://github.com/apache/cordova-plugin-inappbrowser/pull/170/commits/e98085c17d02f252273fada13f28c3b83e21378c)
**Dashboard** - [Link](http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-inappbrowser-pr/55/)

| Builder Name  | Console Output | Test Report | Device Logs  |
|     :---:     |     :---:      |   :---:     |     :---:    |
| [Windows 8.1 Store]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-inappbrowser-pr/55//PLATFORM=windows-8.1-store/) | [Link]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-inappbrowser-pr/55//PLATFORM=windows-8.1-store/console) | [Link]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-inappbrowser-pr/55//PLATFORM=windows-8.1-store/testReport/) | [Link]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-inappbrowser-pr/55//PLATFORM=windows-8.1-store/artifact/) |
| [Windows 10  Store]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-inappbrowser-pr/55//PLATFORM=windows-10-store/) | [Link]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-inappbrowser-pr/55//PLATFORM=windows-10-store/console) | [Link]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-inappbrowser-pr/55//PLATFORM=windows-10-store/testReport/) | [Link]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-inappbrowser-pr/55//PLATFORM=windows-10-store/artifact/) |
| [Windows 8.1 Phone]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-inappbrowser-pr/55//PLATFORM=windows-8.1-phone/) | [Link]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-inappbrowser-pr/55//PLATFORM=windows-8.1-phone/console) | [Link]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-inappbrowser-pr/55//PLATFORM=windows-8.1-phone/testReport/) | [Link]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-inappbrowser-pr/55//PLATFORM=windows-8.1-phone/artifact/) |
| [iOS]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-inappbrowser-pr/55//PLATFORM=ios/) | [Link]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-inappbrowser-pr/55//PLATFORM=ios/console) | [Link]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-inappbrowser-pr/55//PLATFORM=ios/testReport/) | [Link]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-inappbrowser-pr/55//PLATFORM=ios/artifact/) |
| [Android]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-inappbrowser-pr/55//PLATFORM=android/) | [Link]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-inappbrowser-pr/55//PLATFORM=android/console) | [Link]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-inappbrowser-pr/55//PLATFORM=android/testReport/) | [Link]( http://cordova-ci.cloudapp.net:8080/job/cordova-plugin-inappbrowser-pr/55//PLATFORM=android/artifact/) |

**Comment by jcesarmobile** [comment]
Thanks. I forgot one thing, before we can merge it though, you need to sign Apache's Contributor License Agreement (can be done online): http://www.apache.org/licenses/#clas

And if you squash the 2 commits into one that will be perfect.

**Comment by jcesarmobile** [comment]
As it's a small fix I'm going to merge it without the ICLA, but you should consider filling the Apache's Contributor License Agreement http://www.apache.org/licenses/#clas for future contributions

**Pull request closed without merging**